api/v1alpha1/dnsrecord_types.go

```diff
@@ -33,10 +33,19 @@ const HttpsProtocol HealthProtocol = "HTTPS"
 // By default this health check will be applied to each unique DNS A Record for
 // the listeners assigned to the target gateway
 type HealthCheckSpec struct {
-	Endpoint         string          `json:"endpoint,omitempty"`
-	Port             *int            `json:"port,omitempty"`
-	Protocol         *HealthProtocol `json:"protocol,omitempty"`
-	FailureThreshold *int            `json:"failureThreshold,omitempty"`
+	// Should start with "?" or "|", contain only allowed for URLs characters and end with alphanumeric char or "/"
+	// +kubebuilder:validation:Pattern=`^(?:\?|\/)[\w\-.~:\/?#\[\]@!$&'()*+,;=]+(?:[a-zA-Z0-9]|\/){1}$`
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// Only following ports allowed: 80, 443, 1024-49151
```

Comment:
// Port must be either 80, 443 or 1024 and above

Reply:
I kept the upper limit since it is possible to set a port higher but we won't allow it (the range of unregistered ports is not allowed)